mesonbuild/backend/backends.py

```diff
@@ -80,26 +81,31 @@ def __init__(self, build_dir, trees):
         self.trees = trees
 
 class InstallData:
-    def __init__(self, source_dir, build_dir, prefix, strip_bin,
-                 install_umask, mesonintrospect, version):
+    def __init__(self, source_dir: str, build_dir: str, prefix: str,
+                 strip_bin: T.List[str], install_umask: T.Union[str, int],
+                 mesonintrospect: T.List[str], version: str):
+        # TODO: in python 3.8 or with typing_Extensions install_umask could be:
+        # `T.Union[T.Literal['preserve'], int]`, which would be more accurate.
         self.source_dir = source_dir
         self.build_dir = build_dir
         self.prefix = prefix
         self.strip_bin = strip_bin
         self.install_umask = install_umask
-        self.targets = []
-        self.headers = []
-        self.man = []
-        self.data = []
-        self.po_package_name = ''
+        self.targets: T.List[TargetInstallData] = []
+        self.headers: T.List[T.Tuple[str, str, T.Optional['FileMode']]] = []
+        self.man: T.List[T.Tuple[str, str, T.Optional['FileMode']]] = []
+        self.data: T.List[T.Tuple[str, str, T.Optional['FileMode']]] = []
+        self.po_package_name: str = ''
         self.po = []
-        self.install_scripts = []
-        self.install_subdirs = []
+        self.install_scripts: T.List[build.RunScript] = []
+        self.install_subdirs: T.List[T.Tuple[str, str, T.Optional['FileMode'], T.Tuple[T.Set[str], T.Set[str]]]] = []
```

Comment:
Nothing wrong with the annotation, but this tuple should really be a new class for better readability...

Reply:
I made a type alias for this as well as the other ones a bit above this to reduce duplication there.